synapse/util/retryutils.py

```diff
@@ -265,4 +265,6 @@ async def store_retry_timings() -> None:
                 logger.exception("Failed to store destination_retry_timings")
 
         # we deliberately do this in the background.
-        synapse.logging.context.run_in_background(store_retry_timings)
+        run_as_background_process(
+            f"store_retry_timings-{self.destination}", store_retry_timings
+        )
```

Comment:
Including the destination in the name of the background process might not be such a good idea. I think we'll end up with one metric per destination in prometheus?

Reply:
Oh, true. We could possibly put them as tags or something but that still seems unlikely to work well.